internal/osbuild/systemd_journald_stage.go

```diff
@@ -0,0 +1,66 @@
+package osbuild
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type SystemdJournaldStageOptions struct {
+	Filename string                      `json:"filename"`
+	Config   SystemdJournaldConfigDropin `json:"config"`
+}
+
+func (SystemdJournaldStageOptions) isStageOptions() {}
+
+func NewSystemdJournaldStage(options *SystemdJournaldStageOptions) *Stage {
+	return &Stage{
+		Type:    "org.osbuild.systemd-journald",
+		Options: options,
+	}
+}
+
+type SystemdJournaldConfigDropin struct {
+	Journal SystemdJournaldConfigJournalSection `json:"Journal"`
+}
+
+type systemdJournaldConfigDropin SystemdJournaldConfigDropin
+
+func (s SystemdJournaldConfigDropin) MarshalJSON() ([]byte, error) {
+	if s.Journal == (SystemdJournaldConfigJournalSection{}) {
+		return nil, fmt.Errorf("the 'Journal' section is required")
+	}
+	journalSection := systemdJournaldConfigDropin(s)
+	return json.Marshal(journalSection)
+}
+
+// 'Journal' configuration section, at least one option must be specified
+type SystemdJournaldConfigJournalSection struct {
+	// Controls where to store journal data.
+	Storage *string `json:"Storage,omitempty"`
+	// Sets whether the data objects stored in the journal should be
+	// compressed or not. Can also take threshold values.
+	Compress *string `json:"Compress,omitempty"`
+	// Splits journal files per user or to a single file.
+	SplitMode *string `json:"SplitMode,omitempty"`
+	// Max time to store entries in a single file. By default seconds, may be
+	// sufixed with units (year, month, week, day, h, m) to override this.
+	MaxFileSec *string `json:"MaxFileSec,omitempty"`
+	// Maximum time to store journal entries. By default seconds, may be sufixed
+	// with units (year, month, week, day, h, m) to override this.
+	MaxRetentionSec *string `json:"MaxRetentionSec,omitempty"`
+	// Timeout before synchronizing journal files to disk. Minimum 0.
+	SyncIntervalSec *int `json:"SyncIntervalSec,omitempty"`
+	// Enables/Disables kernel auditing on start-up, leaves it as is if
+	// unspecified.
+	Audit *string `json:"Audit,omitempty"`
```

Comment:
Can we get empty lines between each field here?  Not a big issue, I just find it easier to read.